distribution/lib/Standard/Database/0.0.0-dev/src/Internal/SQLite/SQLite_Error_Mapper.enso

```diff
@@ -14,3 +14,13 @@ type SQLite_Error_Mapper
         case error.java_exception of
             sqlite_exception : SQLiteException ->
                 sqlite_exception.getResultCode == SQLiteErrorCode.SQLITE_CONSTRAINT_PRIMARYKEY
+
+    ## PRIVATE
+    transform_custom_errors : SQL_Error -> Any
+    transform_custom_errors error =
+        invariant_failure_message = "unable to use function MATCH in the requested context"
+        message = error.java_exception.getMessage
+        if message.contains invariant_failure_message . not then error else
+            query_contains_invariant_check = error.related_query.if_nothing "" . contains "ENSO INVARIANT VIOLATED"
+            if query_contains_invariant_check . not then error else
+                Invariant_Violation.Error Nothing error
```

Comment:
In SQLite, we are not able to decode the message that our `RUNTIME_ERROR` operation is carrying from the SQL error, like we do in Postgres.

I was looking around and it does not seem possible - I searched the SQLite's source code for usages of `sqlite3_result_error`, but I have found that all usages in the core only rely on constant messages - so there is no hope of 'smuggle' our error message in there.

---

If we really wanted to - we could write a custom C extension to SQLite, defining a function like `raise_error` that would allow us to set such a custom error. We could then compile it on our 3 supported platforms, deliver it together with the `Database` library and load using the SQLite `load_extension` capability.

Of course that seems like an overkill for a feature that is not used very often - after all we expect the invariant violation will happen very rarely - only if another user is doing breaking changes to the DB. And just re-running the workflow should be sufficient to proceed.

But if we ever decide that we want better error reporting in SQLite, such extension could be a way to do it.